cstore_fdw.c

```diff
@@ -254,6 +248,139 @@ cstore_ddl_event_end_trigger(PG_FUNCTION_ARGS)
 
 
 /*
+ * RegisterCStoreTable registers provided cstore table relation id by
+ * adding a new row to pg_cstore_tables.
+ */
+static void
+RegisterCStoreTable(Oid relationId, Oid fileNodeOid)
+{
+	Oid pgCstoreTablesOid = InvalidOid;
+	Relation pgCstoreTables = NULL;
+	TupleDesc tupleDescriptor = NULL;
+	HeapTuple heapTuple = NULL;
+	Datum values[] = { ObjectIdGetDatum(relationId), ObjectIdGetDatum(fileNodeOid) };
+	bool isNulls[] = { false, false };
+
+	pgCstoreTablesOid = get_relname_relid(CSTORE_TABLES_NAME, PG_CATALOG_NAMESPACE);
+
+	if (pgCstoreTablesOid == InvalidOid)
+	{
+		ereport(ERROR, (errmsg("could not find table %s", CSTORE_TABLES_NAME)));
+	}
+
+	/* open shard relation and insert new tuple */
+	pgCstoreTables = heap_open(pgCstoreTablesOid, RowExclusiveLock);
+	tupleDescriptor = RelationGetDescr(pgCstoreTables);
+	Assert(tupleDescriptor->natts == 2);
+	heapTuple = heap_form_tuple(tupleDescriptor, values, isNulls);
+
+#if PG_VERSION_NUM >= 100000
+	CatalogTupleInsert(pgCstoreTables, heapTuple);
+#else
+	simple_heap_insert(pgCstoreTables, heapTuple);
+	CatalogUpdateIndexes(pgCstoreTables, heapTuple);
+#endif
+
+	heap_close(pgCstoreTables, NoLock);
+}
+
+
+/*
+ * UnregisterCStoreTable unregisters a cstore_fdw table by removing
+ * the row from table pg_cstore_tables.
+ */
+static void
+UnregisterCStoreTable(Oid relationId)
+{
+	Oid pgCstoreTablesOid = InvalidOid;
+	Relation pgCstoreTables = NULL;
+	HeapTuple heapTuple = NULL;
+	int scanKeyCount = 1;
+	ScanKeyData scanKey[1];
+	HeapScanDesc scanDescriptor = NULL;
+
+	pgCstoreTablesOid = get_relname_relid(CSTORE_TABLES_NAME, PG_CATALOG_NAMESPACE);
+
```

Comment:
Remove empty line? (What is the style guide for empty lines? What I remember is that when we set a variable and then check for its value we didn't put empty line)